### PR TITLE
Rework ping logic, have channeld ping regularly.

### PR DESCRIPTION
**Pull request by rustyrussell**
This might help stability of Tor connections, but I made it universal since it's easier (in particular, it's hard to tell which *incoming* connections are Tor, since they usually come from localhost).

### CODE REVIEW

File: channeld/channeld.c [hunk 1]
```diff
@@ -2252,6 +2267,19 @@ static void peer_in(struct peer *peer, const u8 *msg)
...
+			handle_ping_reply(peer, msg);
+			/* fall thru */
+		case PONG_EXPECTED_PROBING:
+			peer->expecting_pong = PONG_UNEXPECTED;
+			return;
+		case PONG_UNEXPECTED:
+			status_debug("Unexpected pong?");
+			return;
+		}
+		abort();
```
Comment (vincenzopalazzo) [inline channeld/channeld.c:2282]: Why do we need to about with we receive the `PONG_EXPECTED_COMMAND` status?

File: tests/test_misc.py [hunk 2]
```diff
@@ -254,7 +254,7 @@ def mock_getblock(r):
 
 
 def test_ping(node_factory):
-    l1, l2 = node_factory.line_graph(2, fundchannel=False)
+    l1, l2 = node_factory.line_graph(2)
 
     def ping_tests(l1, l2):
         # 0-byte pong gives just type + length field.
```
Comment (vincenzopalazzo) [inline tests/test_misc.py:260]:
```suggestion
        # 0-byte pong gives just type + length field.
        
        l1.rpc.ping(l2.info['id'])
```

Maybe we can test also if the node responds correctly to the ping that should be run without any problem?

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]
LGTM https://github.com/ElementsProject/lightning/pull/4804/commits/212011def00c402def3cdb3b7f71e3041700ffae

I have only a small comment, that I was not able to answer myself.

File: channeld/channeld.c (hunk 1 above)
Comment (rustyrussell) [reply channeld/channeld.c:2282]: We process the reply, then fall through to the next statement: `peer->expecting_pong = PONG_UNEXPECTED;`

**Comment by cdecker** [comment]
Rebased on `master` since this seems to have tripped up on #4811

**Comment by rustyrussell** [comment]
Rebased again, fixed accidental delete of member initialization.

**Review by vincenzopalazzo — APPROVED** [review]
ack https://github.com/ElementsProject/lightning/pull/4804/commits/e9551cf9b886f8c704b4f34b064f66ae2cef2d7e

**Pull request merged**